src/minisketchwrapper.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <minisketchwrapper.h>
+
+#include <logging.h>
+#include <util/time.h>
+
+#include <minisketch.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <optional>
+#include <utility>
+#include <vector>
+
+namespace {
+
+static constexpr uint32_t BITS = 32;
+
+uint32_t FindBestImplementation()
+{
+    std::optional<std::pair<int64_t, uint32_t>> best;
+
+    uint32_t max_impl = Minisketch::MaxImplementation();
+    for (uint32_t impl = 0; impl <= max_impl; ++impl) {
+        std::vector<int64_t> benches;
+        uint64_t offset = 0;
+        /* Run a little benchmark with capacity 32, adding 184 entries, and decoding 16 of them once. */
+        for (int b = 0; b < 11; ++b) {
+            if (!Minisketch::ImplementationSupported(BITS, impl)) break;
+            Minisketch sketch(BITS, impl, 32);
+            auto start = GetTimeMicros();
+            for (uint64_t e = 0; e < 100; ++e) {
+                sketch.Add(e*1337 + b*13337 + offset);
+            }
+            for (uint64_t e = 0; e < 84; ++e) {
+                sketch.Add(e*1337 + b*13337 + offset);
+            }
+            offset += (*sketch.Decode(32))[0];
```

Comment:
Why use the `optional` version of `Decode` if the value is ignored?

Also, what's the offset's purpose here? It's not obvious to me what will be in `[0]`.